src/tests/ftest/server/daos_server_dump.py

```diff
@@ -25,43 +26,99 @@ def __init__(self, *args, **kwargs):
-        :avocado: tags=vm
-        :avocado: tags=control,server_start,basic
-        :avocado: tags=daos_server_dump_test,test_daos_server_dump_basic
+        :avocado: tags=all
+        :avocado: tags=daos_server_dump_tests,test_daos_server_dump_basic
         """
 
-        ret_codes = stop_processes(self.hostlist_servers, r"daos_engine",
-                           added_filter=r"'\<(grep|defunct)\>'",
-                           dump_ult_stacks=True)
+        ret_codes = dump_engines_stacks(self.hostlist_servers,
+                           added_filter=r"'\<(grep|defunct)\>'")
+        # at this time there is no way to know when Argobots ULTs stacks
+        # has completed, see DAOS-1452/DAOS-9942.
         if 1 in ret_codes:
             print(
-                "Stopped daos_engine processes on {}".format(
+                "Dumped daos_engine stacks on {}".format(
                     str(ret_codes[1])))
         if 0 in ret_codes:
-            print(
+            self.fail(
                 "No daos_engine processes found on {}".format(
                     str(ret_codes[0])))
 
-        # XXX may need to check for one file per engine...
-        ret_codes = pcmd(self.hostlist_servers, r"ls /tmp/daos_dump*.txt")
-        # Report any failures
-        if len(ret_codes) > 1 or 0 not in ret_codes:
-            failed = [
-                "{}: rc={}".format(val, key)
-                for key, val in ret_codes.items() if key != 0
-            ]
-            self.fail(
-                "no ULT stacks dump found on following hosts: {}".format(
-                ", ".join(failed)))
-
         self.log.info("Test passed!")
 
-        # set stopped servers state to make teardown happy
+    def test_daos_server_dump_on_error(self):
+        """JIRA ID: DAOS-1452.
+
+        Test Description: Test engine ULT stacks dump (error case).
+
+        :avocado: tags=all
+        :avocado: tags=daos_server_dump_tests,test_daos_server_dump_on_error
+        """
+
+        self.log.info("Forcing test error!")
+        self.error()
+
+    def test_daos_server_dump_on_fail(self):
+        """JIRA ID: DAOS-1452.
+
+        Test Description: Test engine ULT stacks dump (failure case).
+
+        :avocado: tags=all
+        :avocado: tags=daos_server_dump_tests,test_daos_server_dump_on_fail
+        """
+
+        self.log.info("Forcing test failure!")
+        self.fail()
+
+    def test_daos_server_dump_on_timeout(self):
+        """JIRA ID: DAOS-1452.
+
+        Test Description: Test engine ULT stacks dump (timeout case).
+
+        :avocado: tags=all
+        :avocado: tags=daos_server_dump_tests,test_daos_server_dump_on_timeout
+        """
+
+        self.log.info("Sleeping to trigger test timeout!")
+        time.sleep(30)
+
+    def test_daos_server_dump_on_unexpected_engine_status(self):
+        """JIRA ID: DAOS-1452.
+
+        Test Description: Test engine ULT stacks dump (unexpected engine status case).
+
+        :avocado: tags=all
```

Comment:
Is this a manual test, if so it should have the manual tag:
```suggestion
        :avocado: tags=all,manual
```

Reply:
ok

Reply:
As I mentioned in my main review comment this doesn't have to be addressed in this cherry-pick, but we should clean it up in master at some point.